src/js/utils/JobUtil.js

```diff
@@ -0,0 +1,90 @@
+import Job from '../structs/Job';
+
+const getMatchingProperties = function (job, item) {
```

Comment:
This is getting the model from the schema. `getter` is a attribute on the schema which is an anonymous function which returns the value for that field from the struct. This is used because a) it gives in the schema the possibility to transform the data from the struct to the form. Think key value to array, or array to comma separated list. b) it is a fairly simple approach which you can see directly on the schema. So if a field is filled out wrong there is just one point to look at.

This function is actually walking recursively  trough the schema and creates the model for the form from a struct. The idea is to fill out dynamically the model by giving a schema and a struct in this case the `jobs` struct.